src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -135,53 +169,125 @@ trait GraphDAO {
+  def saveToVertex[T: TypeTag: ClassTag](obj: T, vertex: Vertex, graph: Graph, workspaceName: WorkspaceName): Vertex = {
 
-    def getProperties(obj: T, prefix: String): Iterable[(String, Option[Any])] = {
+    def getProperties(obj: T): Iterable[(String, Option[Any])] = {
       val mirror = ru.runtimeMirror(obj.getClass.getClassLoader)
 
-      for (member <- ru.typeTag[T].tpe.members
-           if (member.asTerm.isVal || member.asTerm.isVar) && member.annotations.exists(_.tree.tpe.typeSymbol == ru.typeOf[VertexProperty@field].typeSymbol)
+      for (member <- ru.typeTag[T].tpe.members if (member.asTerm.isVal || member.asTerm.isVar)
       ) yield {
         val fieldMirror = mirror.reflect(obj).reflectField(member.asTerm)
-        prefix + member.name.decodedName.toString.trim -> (fieldMirror.get match {
+        member.name.decodedName.toString.trim -> (fieldMirror.get match {
           case v: Option[Any] => v
           case v => Option(v)
         })
       }
     }
 
-    getProperties(obj, "_").foreach(_ match {
-      case (key, None) => vertex.removeProperty(key)
+    getProperties(obj).foreach(_ match {
+      case (key, None) =>
+        // can't tell if this is a value or a map anymore so just remove both (there should only be one)
+        vertex.removeProperty(key)
+        vertex.getVertices(Direction.OUT, key).foreach(removeMapVertex)
       case (key, Some(value: DateTime)) => vertex.setProperty(key, value.toDate)
+      case (key, Some(value: Map[_,_])) => serializeAttributeMap(vertex, key, value.asInstanceOf[Map[String, Attribute]], graph, workspaceName)
+      case (key, Some(value: AttributeReference)) => serializeReference(vertex, key, value, graph, workspaceName)
+      case (key, Some(seq: Seq[_])) => seq.headOption match {
+        case None => // empty, do nothing
+        case Some(x: Workflow) =>
+          serializeDomainObjects(vertex, seq.asInstanceOf[Seq[Workflow]], workflowEdge, (w: Workflow) => w.workflowId, graph, workspaceName)
+          val workspaceVertex = getSinglePipelineResult(workspacePipeline(graph, workspaceName.namespace, workspaceName.name)).get
+          new GremlinPipeline(vertex).out(workflowEdge).linkIn(workflowEdge, workspaceVertex).iterate()
+        case Some(x: WorkflowFailure) => serializeDomainObjects(vertex, seq.asInstanceOf[Seq[WorkflowFailure]], workflowFailureEdge, (wf: WorkflowFailure) => (wf.entityType, wf.entityName), graph, workspaceName)
+      }
+      case (key, Some(mrConfig: MethodRepoConfiguration)) => serializeDomainObjects(vertex, Seq(mrConfig), methodRepoConfigEdge, thereShouldOnlyBeOne, graph, workspaceName)
+      case (key, Some(mrMethod: MethodRepoMethod)) => serializeDomainObjects(vertex, Seq(mrMethod), methodRepMethodEdge, thereShouldOnlyBeOne, graph, workspaceName)
+      case (key, Some(ws: WorkspaceName)) => // don't serialize workspace name objects
       case (key, Some(value)) => vertex.setProperty(key, value)
```

Comment:
if none of the cases above catch, this catches all, what happens if we try to set the property to something we don't expect?  for some circumstances we would get orient exceptions, others maybe not and just unexpected behavior.  Maybe we should be explicit about what we support and throw exception for any type not explicitly defined?